src/Native/Runtime/StackFrameIterator.cpp

```diff
@@ -569,7 +569,39 @@ void StackFrameIterator::UnwindFuncletInvokeThunk()
 
     bool isFilterInvoke = EQUALS_CODE_ADDRESS(m_ControlPC, RhpCallFilterFunclet2);
 
-#ifdef _TARGET_AMD64_
+#if defined(UNIX_AMD64_ABI)
+    if (isFilterInvoke)
+    {
+        SP = (PTR_UIntNative)(m_RegDisplay.SP);
+        m_RegDisplay.pRbp = SP++;
+    }
+    else
+    {
+        // Save the preserved regs portion of the REGDISPLAY across the unwind through the C# EH dispatch code.
+        m_funcletPtrs.pRbp = m_RegDisplay.pRbp;
+        m_funcletPtrs.pRbx = m_RegDisplay.pRbx;
+        m_funcletPtrs.pR12 = m_RegDisplay.pR12;
+        m_funcletPtrs.pR13 = m_RegDisplay.pR13;
+        m_funcletPtrs.pR14 = m_RegDisplay.pR14;
+        m_funcletPtrs.pR15 = m_RegDisplay.pR15;
+
+        if (EQUALS_CODE_ADDRESS(m_ControlPC, RhpCallCatchFunclet2))
+        {
+            SP = (PTR_UIntNative)(m_RegDisplay.SP + 0x38);
```

Comment:
Yes, we could declare the abstract operations as inline functions on the StackFrameIterator class and then platform-specific header files could provide the inline function definitions.  It would give us the opportunity to better document the expected semantic at the declaration point and have separated implementations of the methods per-ABI.   It's possible that the current methods (such as UnwindFuncletInvokeThunk / UpdateFromExceptionDispatch / etc) are at the the right abstraction level, but I'm willing to consider that they aren't.